### PR TITLE
Make garbage collection consistent

### PR DESCRIPTION
**Pull request by marcoscaceres**
Cleaned up the other instances...

### CODE REVIEW

File: FileAPI/url/url-with-fetch.any.js [hunk 1]
```diff
@@ -1,26 +1,22 @@
...
-    var gcRec = function (n) {
-      if (n < 1)
-        return {};
-      var temp = {i: "ab" + i + (i / 100000)};
-      temp += "foo";
-      gcRec(n-1);
-    };
-    for (var i = 0; i < 1000; i++)
-      gcRec(10);
+function garbageCollect() {
```
Comment (saschanaz) [inline FileAPI/url/url-with-fetch.any.js:3]: Maybe we should get a common helper function rather than having multiple duplicates?

**Review by domenic — COMMENTED** [review]
Very nice! Approving since this is a strict improvement, but some suggestions:

- Copy the more informative comments from `streams/`
- Use the `self.garbageCollect = () => { }` format since it makes it clearer that it's adding a global.
- Consolidate these all into `/common/gc.js`
- Fix the [others](https://github.com/search?q=repo%3Aweb-platform-tests%2Fwpt%20%22function%20garbageCollect%22&type=code) found in permissions/ and fs/, and [the ones](https://github.com/search?q=repo%3Aweb-platform-tests%2Fwpt+%22function+runGarbageCollection%22&type=code) found in `webusb/` and `bluetooth/`, and finally [the one found in js/builtins](https://github.com/web-platform-tests/wpt/blob/0802d8bd01e17178d9cd2355e6fbaf3c098e7953/js/builtins/weakrefs/resources/maybe-garbage-collect.js#L13).
- Fix [various cases that aren't using this, to use this](https://github.com/search?q=repo%3Aweb-platform-tests%2Fwpt+%22https%3A%2F%2Fgithub.com%2Fweb-platform-tests%2Fwpt%2Fissues%2F7899%29%22&type=code)

**Review by domenic — APPROVED** [review]
Actually approving, as I said I would. To reiterate, I think all my suggestions (and @saschanaz's) are good, but I also know what it's like for test cleanup work to get abandoned because someone suggests EVEN MORE CLEANUP, so I don't want to get us into that situation :)

**Comment by marcoscaceres** [comment]
Let's start with this.

**Pull request merged**